stdlib/utils/src/main/ballerina/utils/built-in-methods.bal

```diff
@@ -19,3 +19,15 @@
 # + value - Value to be cloned
 # + return - Clone of the given value
 extern function clone(anydata value) returns anydata;
+
+# Freeze a given value.
+#
+# + value - Value to be frozen
+# + return - Frozen value
+extern function freeze(anydata value) returns anydata;
```

Comment:
We also allow `.freeze()` and `.isFrozen()` on non-anydata typed variables which can hold `anydata` values.
For example, a `map<any>` that only holds `anydata` values.

Isn't saying `anydata` here an issue?

Reply:
Have we covered them in tests? All the tests got passed. What is the datatype we will use here then?

Reply:
Yeah these are covered, for example:
- https://github.com/ballerina-platform/ballerina-lang/blob/refactor_builtin_functions/tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/builtinfunctions/FreezeAndIsFrozenTest.java#L367 
- https://github.com/ballerina-platform/ballerina-lang/blob/refactor_builtin_functions/tests/ballerina-unit-test/src/test/resources/test-src/expressions/builtinoperations/freeze-and-isfrozen.bal#L399

In such scenarios, the return type is `T|error`, not just `T` too. So the return type could be `any|error` too, not just `any`.

Wonder how this worked in the first place?

Reply:
There is an issue with the test then :) Will fix the method anyway.

Reply:
Ideally compilation should have failed right? According to the log, `freeze-and-isfrozen.bal` seems to have compiled without issues. Maybe the types specified here are not validated?